src/adapters/scene_direction.js

```diff
@@ -26,13 +36,59 @@ const createMarker = (lngLat, className = '', options = {}) => {
 
 const createRouteLabel = (route, vehicle, { lngLat, anchor }) => {
   const element = document.createElement('div');
-  element.innerHTML = `
-    <div class="routeLabel-vehicleIcon ${getVehicleIcon(vehicle)}"></div>
+  let content;
```

Comment:
This works to use the JSX syntax here, but this doesn't bring much simplification in this file which is already big.
What I meant was creating a real new React component in a separate file, for example `RouteLabel.jsx`, so you can focus only on what the label renders looks like.
And here, you can call our `renderStaticReact` function to get the string representation to inject as your element `innerHTML`.